### DISCUSSION TITLE
How to find a Type anywhere in the workspace?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-jdt/eclipse.jdt.core#522] [Q&A] [answered]
Currently I have some code that do the following:

```
IJavaProject project = ...
IType type = project.findType("my.type.ToFind", subMonitor.split(10))
```

But now I want to do the reverse, having a type name ("my.type.ToFind") and find all `IJavaProject` that contains such a named type.

It seems the "Open Declaration" Action is doing something similar  as even if the type is not available on the classpath it shows me where it is located (or opens a dialog if that is not unique) and also "Find Type" obviously do something like this (but I only want exact matches). 

I could think about iterating all projects in the workspace and try each one with the `findType` method, but wonder if there is a "smarter" way of doing this.

**Comment by szarnekow** [comment, marked as answer]
I think the SearchEngine is what you are looking for. It allows to search for IType and other Java elements.

**Reply by laeubi** [reply]
Thanks this looks very promising I'll investigate on this more!
